api/v1_projects.inc

```diff
@@ -880,6 +880,37 @@ function api_v1_project_page(string $method, array $data, array $query_params)
     }
 }
 
+function api_v1_project_page_report_bad(string $method, array $data, array $query_params): void
+{
+    global $pguser, $PAGE_BADNESS_REASONS;
+
+    try {
+        $project = $data[":projectid"];
+        $proof_project = new ProofProject($project);
+
+        $project_page = $data[":pagename"];
+        $proof_project_page = new ProofProjectPage($proof_project, $project_page);
+
+        $reason_string = receive_data_from_request_body("reason");
+        // can't use array_find_key() yet
```

Comment:
Should we update the symfony polyfil so we can use this function? then when we move to php8.* it will be ready.

Reply:
This function is in 8.4 so that's a ways out from our minimum version (8.2 going forward IIRC). But yes, I'm fine updating our polyfills to include the 8.x ones so we can use this. Let's do that in a separate PR.

Reply:
Presumably I can't do that because it is not under git control.

Reply:
It's under git: https://github.com/DistributedProofreaders/dproofreaders/pull/1423